hack/observability/visualizer/values.yaml

```diff
@@ -3,9 +3,8 @@ replicas: 1
 image:
   repository: ghcr.io/jont828
   name: cluster-api-visualizer
+  tag: latest # TODO: do we want to pin this to latest?
```

Comment:
@fabriziopandini One way we can use the latest version of the chart is to pin the tag to `latest` in the values file. I did notice an issue where we aren't fetching the latest version of the chart if the directory `hack/observability/visualizer/charts` already exists. If we change the tilt-prepare logic to always fetch the latest version of the Helm chart instead, we could keep it up to date that way.

Reply:
I would prefer to not have tilt-prepare to treat the chart as a black box, so let's keep `tag: latest` in the values file

Reply:
Sounds good to me. One thing I want to test on my end is if tilt-prepare will fetch the latest version of the chart if `hack/observability/visualizer/charts/cluster-api-visualizer` folder exists with an older version.

Reply:
I think kustomize does not pull a chart again if a chart already exists locally and it looks like any changes to kustomization.yaml (or values.yaml) does not trigger a re-pull. I hit a similar issue when working with other observability tools. 

From Kustomize documentation:
> If a chart exists, kustomize will not overwrite it (so to suppress a pull, simply assure the chart is already in your kustomization root). kustomize won't check dates or version numbers or do anything that smells like cache management.

With this it looks like each person might be using a different version of the chart depedning on when they pulled it for the first time.

One way to work around this might be to clean up the `charts` folder under each of the enabled observability tools (eg: delete `hack/observability/visualizer/charts`) in tilt-prepare before the kustomize task is run. This should force a re-pull.



cc @killianmuldoon This is similar to the issue we hit with promtail.

Reply:
That would add to the set-up time, but we could make it configurable somewhere. Another option - I'll also state this on the other issue - might be to drop the helm charting and install directly from manifests.

Reply:
If we can delete and repull the charts that would allow us to drop the `tag: latest` in the values file for the visualizer too. If we aren't pulling the latest chart and we pin the image version to `latest`, it can introduce an edge case where a newer version of the image isn't compatible with an older version of the chart.

Reply:
Might be a stupid question but why do we want to use latest? In general it is considered an anti-pattern to use latest as your deployment will just change if someone pushes a new version. (and as @Jont828 mentioned, it can happen that the Helm chart doesn't work with the new image).

I checked the other charts we are using and there is only one place in the Grafana Chart where latest is used (but not for the main grafana image). All other charts are using pinned tags.

Reply:
>In all other places pinned tags are used.

Probably a separate discussion: Looks like the images for a particular chart version are pinned but our `kustomization.yaml` for the observability tools do not pin a chart version. For a user, depnending on when they run `tilt up` for the first time it be pulling down a different version of the chart (and potentially a differente version of the component?). I have observed this locally. I had a discussion with @killianmuldoon about this and I will be openning an issue to pin these versions. 

Dropping this here as it seems kinda relevant.

Reply:
You could argue that for dev it's fine to just use the latest charts, especially as we don't really do any testing for our tilt dev env with those charts and whatever they publish upstream should be fine for us.

But it's definitely a more controlled way to pin it all, so no objections from my side.

Reply:
Based on what we saw from the kustomize docs, it looks like there isn't a great way to pull the latest version of the chart. The options I can think of for now are to (1) add a flow for cleaning up existing charts, i.e. have `make clean` remove the local copy of charts` or (2) modify the tilt-prepare logic to pin it to a version. Do you have any thoughts on how we can proceed?

Reply:
I would favor an additional `make clean-tilt` (exact name TBD) target which deletes the charts. I think considering we're talking about a dev env this should be fine for now.

Reply:
Regarding the values.yaml here. I would drop all image related fields so we use them from the chart (then we always take the version that is published in the corresponding Chart version)